contents/docs/how-posthog-works/ingestion-pipeline.mdx

```diff
@@ -26,6 +26,11 @@ graph TD
 
     Kafka2 <-..- ClickHouse
 ```
+## Why am I seeing duplicate events?
+
+We recommend sending a UUID with every request, and some of our libraries such as `posthog-js` will do this automatically. If you don't send UUIDs for every event, we wouldn't be able to know it's the same event and hence we wouldn't be able to unduplicate it for you. Events with the same uuid, event name, timestamp and distinct_id are considered duplicates and will be eventually unduplicated. Keep in mind that failures and retries happen, so your application or our library might send the same event multiple times, which if you don't set UUIDs would end up in PostHog as separate events.
+
+If you're using a library which doesn't currently support UUIDs for every event, please consider creating a PR or filing an issue in GitHub.
```

Comment:
```suggestion
Most of our SDKs, such as JavaScript Web, Python, and our API, do this automatically. If you're using an SDK which doesn't currently support UUIDs for every event, please consider creating a PR or filing an issue in GitHub.
```

Reply:
I think python doesn't do it automatically ... I could be wrong ... but you can supply it to capture in python lib https://github.com/PostHog/posthog-python/blob/139258cacb7956f1bc025ad9c42095125bd6cd81/posthog/__init__.py#L31

Reply:
In my instance both Python and the API automatically did it.

API:
![Screen Shot 2023-09-15 at 10 44 33 AM](https://github.com/PostHog/posthog.com/assets/34755028/9b06e438-82cc-4adc-9f7f-ec110a09aa4d)

Python:
![Screen Shot 2023-09-15 at 10 45 15 AM](https://github.com/PostHog/posthog.com/assets/34755028/709be260-15e6-4c27-ade6-e641238f2035)

Reply:
We add uuid at capture, if it wasn't sent & that's the problem

Reply:
Ahhh, didn't know that. You're right then. It doesn't look like Python is sending the UUID.

Reply:
```suggestion
Some of our SDKs, such as JavaScript Web, do this automatically, others allow you to pass UUIDs. If you're using an SDK which doesn't currently support UUIDs for every event, please consider creating a PR or filing an issue in GitHub.
```

Reply:
@joethreepwood  ^ this change & then ✅